compiler/next/lib/parsing/chpl.ypp

```diff
@@ -2983,31 +2957,63 @@ expr:
 | TLP TDOTDOTDOT expr TRP
     { $$ = TODOEXPR(@$); }
 | expr TCOLON expr
-    { $$ = TODOEXPR(@$); }
+  {
+    $$ = context->buildBinOp(@$, $1, $2, $3);
+  }
 | expr TDOTDOT expr
-    { $$ = TODOEXPR(@$); }
+  {
+    $$ = Range::build(BUILDER, LOC(@$), Range::DEFAULT, toOwned($1),
+                      toOwned($3)).release();
+  }
 | expr TDOTDOTOPENHIGH expr
-    { $$ = TODOEXPR(@$); }
+  {
+    $$ = Range::build(BUILDER, LOC(@$), Range::OPEN_UPPER, toOwned($1),
```

Comment:
I see that we've created an inconsistency between "open high" and "open upper" here. FWIW. "high" is a more consistent term considering the range interface. I'm OK if this is fixed later.

Reply:
I will revert back to `OPEN_HIGH`. It was `CLOSED` that was the real problem, and that's `DEFAULT` now.